src/jdk.compiler/share/classes/com/sun/tools/javac/parser/JavacParser.java

```diff
@@ -2043,22 +2063,30 @@ enum LambdaParameterKind {
     };
 
     class LambdaClassifier {
-
         LambdaParameterKind kind;
         Fragment diagFragment;
-        List<JCVariableDecl> params;
 
+        /**
+         * analyzeParens() has already classified the lambda as EXPLICIT_LAMBDA, due to
+         * two consecutive identifiers. adding an erroneous parameter (one that
```

Comment:
Shouldn't the second sentence say something more like:
```
Because of that (<explicit lambda>), the parser will always attempt to parse a type, followed by a name. If the lambda contains an illegal mix of implicit and explicit parameters, it is possible for the parser to see a `,` when expecting a name, in which case the variable is created with an erroneous name. The logic below makes sure that the lambda parameters are all declared with either an explicit type (e.g. `String x`), or with an inferred type (using `var x`). Any other combination is rejected.
```
*